### ISSUE TITLE
Add support for Unicode properties `Script` values added in ES2022

### ISSUE THREAD

**Issue by ota-meshi** [eslint/eslint#15772]
### Syntax name

List new Unicode v14 `Script`/`Script_Extensions` values

### Syntax proposal URL

There are no proposals. These seem to have been added due to the Unicode v14 update.

https://github.com/tc39/ecma262/pull/2515

### Example code

```js
/* eslint no-invalid-regexp : ["error"] */

// False positives
new RegExp("\\p{Script=Cpmn}", "u");
new RegExp("\\p{Script=Cypro_Minoan}", "u");
new RegExp("\\p{Script=Old_Uyghur}", "u");
new RegExp("\\p{Script=Ougr}", "u");
new RegExp("\\p{Script=Tangsa}", "u");
new RegExp("\\p{Script=Tnsa}", "u");
new RegExp("\\p{Script=Toto}", "u");
new RegExp("\\p{Script=Vith}", "u");
new RegExp("\\p{Script=Vithkuqi}", "u");
```

### Implementation Checklist

- [X] Ecma262 update: https://github.com/tc39/ecma262/pull/2515
- [X] Acorn update: https://github.com/acornjs/acorn/pull/1084
- [X] `espree` update: 
- [ ] `regexpp` update: https://github.com/mysticatea/regexpp/pull/27
- [ ] `eslint` update: 

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

I think that can be fixed by changing regexpp.
https://github.com/mysticatea/regexpp/pull/27

ESLint has a false positive in the no-invalid-regexp rule.
[DEMO]

[DEMO]:  https://play.eslint.org/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWludmFsaWQtcmVnZXhwIDogW1wiZXJyb3JcIl0gKi9cblxuLy8gTkdcbm5ldyBSZWdFeHAoXCJcXFxccHtTY3JpcHQ9Q3Btbn1cIiwgXCJ1XCIpO1xubmV3IFJlZ0V4cChcIlxcXFxwe1NjcmlwdD1DeXByb19NaW5vYW59XCIsIFwidVwiKTtcbm5ldyBSZWdFeHAoXCJcXFxccHtTY3JpcHQ9T2xkX1V5Z2h1cn1cIiwgXCJ1XCIpO1xubmV3IFJlZ0V4cChcIlxcXFxwe1NjcmlwdD1PdWdyfVwiLCBcInVcIik7XG5uZXcgUmVnRXhwKFwiXFxcXHB7U2NyaXB0PVRhbmdzYX1cIiwgXCJ1XCIpO1xubmV3IFJlZ0V4cChcIlxcXFxwe1NjcmlwdD1UbnNhfVwiLCBcInVcIik7XG5uZXcgUmVnRXhwKFwiXFxcXHB7U2NyaXB0PVRvdG99XCIsIFwidVwiKTtcbm5ldyBSZWdFeHAoXCJcXFxccHtTY3JpcHQ9Vml0aH1cIiwgXCJ1XCIpO1xubmV3IFJlZ0V4cChcIlxcXFxwe1NjcmlwdD1WaXRoa3VxaX1cIiwgXCJ1XCIpO1xuXG4vLyBPS1xuL1xccHtTY3JpcHQ9Q3Btbn0vdTtcbi9cXHB7U2NyaXB0PUN5cHJvX01pbm9hbn0vdTtcblxuXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbIm9mZiJdLCJmb3ItZGlyZWN0aW9uIjpbIm9mZiJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWV4LWFzc2lnbiI6WyJlcnJvciJdLCJuby1leHRyYS1ib29sZWFuLWNhc3QiOlsiZXJyb3IiXSwibm8tZXh0cmEtc2VtaSI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW5uZXItZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWludmFsaWQtcmVnZXhwIjpbImVycm9yIl0sIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjpbImVycm9yIl0sIm5vLWxvc3Mtb2YtcHJlY2lzaW9uIjpbImVycm9yIl0sIm5vLW1pc2xlYWRpbmctY2hhcmFjdGVyLWNsYXNzIjpbImVycm9yIl0sIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6WyJlcnJvciJdLCJuby1uZXctc3ltYm9sIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC12YXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl0sImFjY2Vzc29yLXBhaXJzIjpbImVycm9yIl19LCJlbnYiOnsiZXM2Ijp0cnVlfX19

**Comment by mdjermanovic** [comment]
@ota-meshi thanks! Marked as accepted. It seems we're just waiting for `regexpp`.

**Comment by nzakas** [comment]
@ota-meshi any idea if regexpp will land this update?

**Comment by ota-meshi** [comment]
@nzakas I will try to contact him after the community RFC is in the final stages.
However, I do not know if I will be able to reach him 😓.

**Comment by nzakas** [comment]
Okay, then we should decide how we want to proceed here. Perhaps we can find a different utility?

**Comment by ota-meshi** [comment]
Do you want this issue to be resolved before the eslint-community RFC is finished? https://github.com/eslint/rfcs/pull/91
If so, I will try to contact mysticatea, regardless of the status of the eslint-community RFC.

> Perhaps we can find a different utility?

I think regexpp is the most compatible lib for ESLint.
Other than that, I think regjsparser is good, but it seems to handle esnext syntax. Therefore, I think it may be difficult to incorporate it into the no-invalid-regexp rule.
https://www.npmjs.com/package/regjsparser

**Comment by nzakas** [comment]
RFCs are open for a minimum of 28 days, so there’s still time to go on that. If you’d rather focus on the RFC, that’s fine, but I don’t think we should assume that it blocks this.

**Comment by ota-meshi** [comment]
> but I don’t think we should assume that it blocks this.

Yeah. I don't think it's blocking.
For now, I think the community RFC is in the not too distant future, so I just want to contact mysticatea once or a few times rather than contacting it many times. I don't want him to be busier than he is now.

**Comment by ota-meshi** [comment]
To make progress in this situation, I think that we need to consider whether to create a fork project for regexpp or use a different package.
(Or should we also consider freezing regex rules?)
Do you have any package ideas to consider other than regjsparser? Also, what do you think about using regjsparser?

**Comment by nzakas** [comment]
I don’t have any particular knowledge or preference as to other packages to use.
